include/json/reader.h

```diff
@@ -200,7 +203,7 @@ class JSON_API Reader {
   bool readCStyleComment();
   bool readCppStyleComment();
   bool readString();
-  void readNumber();
+  bool readNumber(bool checkInf);
```

Comment:
Any tests for this code? How do you know that it works?

Reply:
The code should have been tested in `class OurReader`, it has changed a lot after exacting copy of Reader, renamed to OurReader, I just want to keep them consistent. Or we can just remove the comment `exact copy of Reader, renamed to OurReader`, because it is no longer accurate, and can be misleading.

Reply:
+1 on removing the comment. We should also work on getting all the test coverage on OurReader. Long term I would like to delete the Reader and Writer classes (probably as part of 2.0?)

Reply:
Noticed that you have removed that comment, so do we still need this PR? And yes, delete the deprecated methods can keep the code clean.